include/seastar/rpc/rpc.hh

```diff
@@ -383,6 +385,23 @@ public:
         _outgoing_queue_ready = dummy->done.get_future();
         (void)p.get_future().then([dummy = std::move(dummy)] { dummy->done.set_value(); });
     }
+
+    // Each RPC connection has one such worker fiber, whose sole purpose
+    // is to eventually send an empty RPC frame over the connection whenever
+    // the compressor demands it (by signalling a condition variable).
+    //
+    // This is a mechanism which allows the compressor to communicate
+    // something to the other end of the connection (e.g. to re-negotiate
+    // some parameters of compression, such as a dictionary).
+    future<> spawn_compressor_bumper() {
+        return seastar::repeat([this] {
+            return _compressor_needs_progress.wait().then([this] () {
+                return send({})
+                    .handle_exception([] (const auto&) {})
+                    .then([] () { return stop_iteration::no; });
+            });
+        }).handle_exception_type([] (const broken_condition_variable&) {});
```

Comment:
Note: since 5d3ee980730b49c2379a2728d69698ee68f5dbeb it's permissible (and encouraged) to use coroutines where they make sense.